test/src/main/java/org/axonframework/test/fixture/AxonTestFixture.java

```diff
@@ -0,0 +1,594 @@
+        }
+
+        public Customization registerFieldFilter(FieldFilter fieldFilter) {
+            this.fieldFilters.add(fieldFilter);
+            return this;
+        }
+
+        public Customization registerIgnoredField(Class<?> declaringClass, String fieldName) {
+            return registerFieldFilter(new IgnoreField(declaringClass, fieldName));
+        }
+    }
+
+    /**
+     * Every given method invocation spawn new unit of work.
+     */
+    static class Given implements AxonTestPhase.Given {
+
+        private final NewConfiguration configuration;
+        private final Customization customization;
+        private final RecordingCommandBus commandBus;
+        private final RecordingEventSink eventSink;
+        private final MessageTypeResolver messageTypeResolver;
+        private final List<AsyncUnitOfWork> unitsOfWork = new ArrayList<>();
+
+        Given(
+                NewConfiguration configuration,
+                Customization customization,
+                RecordingCommandBus commandBus,
+                RecordingEventSink eventSink,
+                MessageTypeResolver messageTypeResolver
+        ) {
+            this.configuration = configuration;
+            this.customization = customization;
+            this.commandBus = commandBus;
+            this.eventSink = eventSink;
+            this.messageTypeResolver = messageTypeResolver;
+        }
+
+        @Override
+        public AxonTestPhase.Given noPriorActivity() {
+            return this;
+        }
+
+        @Override
+        public AxonTestPhase.Given event(Object payload, MetaData metaData) {
+            var eventMessage = toGenericEventMessage(payload, metaData);
+            return events(eventMessage);
+        }
+
+        private GenericEventMessage<Object> toGenericEventMessage(Object payload, MetaData metaData) {
+            var messageType = messageTypeResolver.resolve(payload);
+            return new GenericEventMessage<>(
+                    messageType,
+                    payload,
+                    metaData
+            );
+        }
+
+        @Override
+        public AxonTestPhase.Given events(List<?>... events) {
+            var messages = Arrays.stream(events)
+                                 .map(e -> e instanceof EventMessage<?> message
+                                         ? message
+                                         : toGenericEventMessage(e, MetaData.emptyInstance())
+                                 ).toArray(EventMessage<?>[]::new);
+            return events(messages);
+        }
+
+        @Override
+        public AxonTestPhase.Given events(EventMessage<?>... messages) {
+            inUnitOfWorkRunOnInvocation(processingContext -> eventSink.publish(processingContext,
+                                                                               messages));
+            return this;
+        }
+
+        private AsyncUnitOfWork inUnitOfWorkRunOnInvocation(Consumer<ProcessingContext> action) {
```

Comment:
The result of this method, as well as of `inUnitOfWorkOnInvocation`, don't seem to be used. Am I missing something obvious or is this some leftover perhaps?